A couple small GameLoop changes

sukibaby opened the pull request:
This PR pretty much finishes up the rest of the small optimizations I wanted to implement in GameLoop. It's an assortment of smaller commits that wouldn't necessarily warrant their own PR individually.

sukibaby committed:
Optimize avoiding CheckGameLoopTimerSkips

Implements a static boolean to track the CheckGameLoopTimerSkips preference, so that it isn't calling PREFSMAN via CheckGameLoopTimerSkips in a tight loop. Since this setting is rarely used, and isn't expected to be changed while the game is open, it doesn't need to be continuously checked.

florczakraf commented on src/GameLoop.cpp:
```diff
@@ -328,13 +329,13 @@ void GameLoop::RunGameLoop()
+		/* uint8_t's max value is 255, so we will
+		 * check input devices every 255 frames. */
+		static uint8_t delayerCounter2 = 0;
+		if (++delayerCounter2 == 0)
```
~~UB when you overflow an unsigned type~~ I was wrong, it's UB for **signed** types, this is well defined.

florczakraf commented on src/GameLoop.cpp:
```diff
@@ -273,7 +270,13 @@ void GameLoop::UpdateAllButDraw(bool bRunningFromVBLANK)
+	/* uint8_t's max value is 255, so we will
+	 * check input devices every 255 frames. */
+	static uint8_t delayerCounter1 = 0;
+	if (++delayerCounter1 == 0)
```
~~UB when you overflow an unsigned type~~ I was wrong, it's UB for signed types, this is well defined.

florczakraf reviewed (changes requested):
I'm not sure if we need `CheckGameLoopTimerSkips` at all since it's a debug feature that needs to be enabled explicitly via `Preferences.ini` and it only would make sense on underspecced machine with vsync enabled. However, the changes in `UpdateAllButDraw` remove the sole purpose of this function, because it will be unable to report deviations from the expected stable framerate, because it wouldn't run on every paced tick

sukibaby commented:
> I'm not sure if we need `CheckGameLoopTimerSkips` at all since it's a debug feature that needs to be enabled explicitly via `Preferences.ini` and it only would make sense on underspecced machine with vsync enabled.

Honestly I agree. Hardware is generally much better than it was when `CheckGameLoopTimerSkips` was implemented too. New push does a better job of not calling it. :)

> However, the changes in `UpdateAllButDraw` remove the sole purpose of this function, because it will be unable to report deviations from the expected stable framerate, because it wouldn't run on every paced tick

I see that I was focused on avoiding redundant updates when already running from VBLANK, but did not consider that the flag might not be cleared when running from VBLANK.   I accept this.

sukibaby commented:
Changes made since the initial PR:

1. Remove the `PREFSMAN->m_bLogSkips` check from inside `CheckGameLoopTimerSkips`.

2. Implement a static boolean to determine the state of `PREFSMAN->m_bLogSkips` a single time, instead of on each individual frame.

3. Use that boolean to avoid calling `CheckGameLoopTimerSkips` in the first place.

teejusb commented on src/GameLoop.cpp:
```diff
@@ -247,6 +244,9 @@ void GameLoop::UpdateAllButDraw(bool bRunningFromVBLANK)
 	static bool m_bUpdatedDuringVBLANK = false;
+	
+	// Assume this preference isn't going to change while the game is running.
+	static bool bLogSkips = PREFSMAN->m_bLogSkips;
```
Let's put this declaration closer to its usage.

sukibaby committed:
Simplify the input device check

Reduce the input device check from every 500 frames to every 255 frames by way of replacing the modulo 500 with a wrapping uint8_t. A bit faster reaction during device plug-in is nice, and using a simple pre-increment instead of doing a modulo is a little more efficient.
